File: core/domain/entities/ColorFill.ts
import { Color } from './Color';
import { Gradient } from './Gradient';

export enum FillType {
  fill = 0,
  gradient = 1,
  image = 4,
  noise = 5,
}

export const FillTypeValues: number[] = [
  FillType.fill,
  FillType.gradient,
  FillType.image,
  FillType.noise,
];

export class ColorFill {
  isEnabled: boolean;
  color: Color;
  opacity: number; // between 0 to 1 with floating points
  fillType: FillType;
  gradient?: Gradient;
}

File: templates/ios/XcodeProjectTemplate/projectName/DesignToCode/Codables/ColorFill.swift
// To parse the JSON, add this file to your project and do:
//
//   let colorFill = try? newJSONDecoder().decode(ColorFill.self, from: jsonData)

import Foundation
import UIKit

enum FillType: Int, Codable {
    case fill = 0
    case gradient = 1
    case image = 4
    case noise = 5
}

enum GradientType: Int, Codable {
    case linear = 0, radial, angular
}

struct ColorFill: Codable {
    let isEnabled: Bool
    let fillType: FillType
    let opacity: CGFloat
    let color: Color
    let gradient: Gradient?
}

struct Gradient: Codable {
    let elipseLength: CGFloat
    let from, to: Point
    let type: GradientType
    let stops: [Stop]
}

struct Stop: Codable {
    let position: CGFloat
    let color: Color
}
